src/metpy/calc/tools.py

```diff
@@ -1,1620 +1,1660 @@
+        return tuple(indexer if slice_dim % ndims == i else slice(None)  # noqa: S001
+                     for i in range(ndims))
+    return take
+
+
+@exporter.export
+@preprocess_xarray
+def lat_lon_grid_deltas(longitude, latitude, y_dim=-2, x_dim=-1, **kwargs):
+    r"""Calculate the actual delta between grid points that are in latitude/longitude format.
+
+    Parameters
+    ----------
+    longitude : array_like
+        array of longitudes defining the grid. If not a `pint.Quantity`, assumed to be in
+        degrees.
+    latitude : array_like
+        array of latitudes defining the grid. If not a `pint.Quantity`, assumed to be in
+        degrees.
+    y_dim : int
+        axis number for the y dimesion, defaults to -2.
+    x_dim: int
+        axis number for the x dimension, defaults to -1.
+    kwargs
+        Other keyword arguments to pass to :class:`~pyproj.Geod`
+
+    Returns
+    -------
+    dx, dy:
+        at least two dimensional arrays of signed deltas between grid points in the x and y
+        direction
+
+    Notes
+    -----
+    Accepts 1D, 2D, or higher arrays for latitude and longitude
+    Assumes [..., Y, X] dimension order for input and output, unless keyword arguments `y_dim`
+    and `x_dim` are otherwise specified.
+
+    """
+    from pyproj import Geod
+
+    # Inputs must be the same number of dimensions
+    if latitude.ndim != longitude.ndim:
+        raise ValueError('Latitude and longitude must have the same number of dimensions.')
+
+    # If we were given 1D arrays, make a mesh grid
+    if latitude.ndim < 2:
+        longitude, latitude = np.meshgrid(longitude, latitude)
+
+    # pyproj requires ndarrays, not Quantities
+    try:
+        longitude = longitude.m_as('degrees')
+        latitude = latitude.m_as('degrees')
+    except AttributeError:
+        longitude = np.asarray(longitude)
+        latitude = np.asarray(latitude)
+
+    # Determine dimension order for offset slicing
+    take_y = make_take(latitude.ndim, y_dim)
+    take_x = make_take(latitude.ndim, x_dim)
+
+    geod_args = {'ellps': 'sphere'}
+    if kwargs:
+        geod_args = kwargs
+
+    g = Geod(**geod_args)
+
+    forward_az, _, dy = g.inv(longitude[take_y(slice(None, -1))],
+                              latitude[take_y(slice(None, -1))],
+                              longitude[take_y(slice(1, None))],
+                              latitude[take_y(slice(1, None))])
+    dy[(forward_az < -90.) | (forward_az > 90.)] *= -1
+
+    forward_az, _, dx = g.inv(longitude[take_x(slice(None, -1))],
+                              latitude[take_x(slice(None, -1))],
+                              longitude[take_x(slice(1, None))],
+                              latitude[take_x(slice(1, None))])
+    dx[(forward_az < 0.) | (forward_az > 180.)] *= -1
+
+    return dx * units.meter, dy * units.meter
+
+
+@exporter.export
+@preprocess_xarray
+def azimuth_range_to_lat_lon(azimuths, ranges, center_lon, center_lat, **kwargs):

```

Comment:
We usually use singular names for our arguments, so would be more consistent with the rest of the code to use "azimuth" and "range".

Reply:
The editor I was using had the word "range" highlighted in a different color like it was a reserved namespace or something, which is why I changed it to prevent any conflicts. However, if you don't think this is an issue, I can change it back.

Reply:
I just thought of that as well. While not a reserved word, it is the name of a built-in function which we don't want to clash with. And we don't really want (in the function signature) to abbreviate to rng...*sigh* 

Alright, I take it back. :wink: Leave them as is.